### DISCUSSION TITLE
STM32: FIFO for Serial

### DISCUSSION THREAD

**Discussion by ser-plu** [modm-io/modm#1041] [Q&A] [answered]
Hi, guys

I wanted to use the standard UART driver for STM32G431, and I have found that there is no implementation write/read buffer, and it even does not use the hardware FIFOs. Is it correct or do I miss something? How would you suggest then to send/receive long data via serial? Do you have any templates?

**Comment by salkinium** [comment, marked as answer]
The buffer is disabled by default, you may enable it via lbuild options `buffer.tx`, `buffer.rx` for each instance individually. I'm not certain if the driver uses hardware FIFO though.

**Comment by ser-plu** [comment]
Thanks. The `buffer` option is what I was looking for.
